extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/graal/MongoClientSubstitutions.java

```diff
@@ -173,3 +173,13 @@ public String resolveAdditionalQueryParametersFromTxtRecords(final String host)
         throw new UnsupportedOperationException("mongo+srv:// not supported in native mode");
     }
 }
+
+@TargetClass(com.sun.jna.Native.class)
+final class JnaNativeSubstitutions {
```

Comment:
Since this substitution targets a class which isn't MongoDB specific, does it belong here?

My concern is that if such subtitutions are kept within an extension, then they will eventually conflict with other extensions.

I guess it could be acceptable as a temporary solution, if there's an issue tracking a further improvement.

Reply:
We usually put that ones in core considering they are targetting JDK classes.

Reply:
This isn't a JDK class however. It's coming from this dependency: https://github.com/quarkusio/quarkus/blob/6a612a83b9caa1bd99f5ca3a106983ddfe71b206/bom/runtime/pom.xml#L1786